README.md

```diff
@@ -63,6 +63,41 @@ Note: x64 Ruby for Windows is known to have some compatibility issues.
 
 # Getting Started with Azure Resource Manager Usage (Preview)
 
+## Prerequisite
+
+In order to use the Azure SDK, you must supply the following values to the Azure SDK:
+
+* Tenant Id
+* Client Id
+* Subscription Id
+* Client Secret
+
+You could pass the above values in the following ways:
+
+### Option 1 - Environment Variables
+You can set the (above) values using the following environment variables:
+
+* AZURE_TENANT_ID
+* AZURE_CLIENT_ID
+* AZURE_SUBSCRIPTION_ID
+* AZURE_CLIENT_SECRET
+
+### Option 2 - Options Hash
+You can set the (above) values using the options hash:
+
+```ruby
+options = {
+  tenant_id: 'YOUR TENANT ID',
+  client_id: 'YOUR CLIENT ID',
+  client_secret: 'YOUR CLIENT SECRET',
+  subscription_id: 'YOUR SUBSCRIPTION ID'
+}
+```
+
+### Option 3 - Combination of Environment Variables & Options Hash
```

Comment:
what's the recommendation for the case in which the user wants to create credentials themselves? either from TokenProvider or MSITokenProvider? if they were to pass them in the hash, they would still need to pass in the 4 values specified above? so if credentials are passed, there are 5 values required for the hash?

Reply:
Done